src/persist-client/src/read.rs

```diff
@@ -516,6 +659,56 @@ where
         Ok(iter)
     }
 
+    /// Returns a `HollowBatch` enriched with this metadata appropriate for
+    /// snapshot-style behavior.
+    #[instrument(level = "debug", skip_all, fields(shard = %self.machine.shard_id()))]
+    pub async fn snapshot_hollow_batch(
```

Comment:
ditto let's replace the old codepaths, the new way is strictly better! concretely: replace `ReadHandle::snapshot`'s implementation with this (but keep the name `snapshot`) plz. then delete SnapshotShard and SnapshotIter entirely

Reply:
ditto on follow-up PR